src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,755 @@
+	 */
+	public void setStreamStatus(String streamStatus) {
+		this.streamStatus = streamStatus;
+	}
+
+	/**
+	 * This method is called by Symphony to control the properties in the device
+	 *
+	 * @param controllableProperty ControllableProperty instance
+	 * @throws Exception if unexpected the value
+	 */
+	@Override
+	public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+		// To do
+	}
+
+	/**
+	 * This method is called by Symphony to control the properties in the device
+	 *
+	 * @param list the list ControllableProperty instance
+	 * @throws Exception if ControllableProperty list is empty
+	 */
+	@Override
+	public void controlProperties(List<ControllableProperty> list) throws Exception {
+		// To do
+	}
+
+	/**
+	 * This method is called by Symphony to get the list of statistics to be displayed
+	 *
+	 * @return List<Statistics> This return the list of statistics
+	 */
+	@Override
+	public List<Statistics> getMultipleStatistics() throws Exception {
+		ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+		Map<String, String> stats = new HashMap<>();
+		failedMonitor = new HashMap<>();
+		authenticationCookie = initAuthenticationCookie();
+
+		populateDecoderMonitoringMetrics(stats);
+		extendedStatistics.setStatistics(stats);
+
+		return Collections.singletonList(extendedStatistics);
+	}
+
+	@Override
+	protected void authenticate() throws Exception {
+	}
+
+	/**
+	 * @return HttpHeaders contain cookie for authorization
+	 */
+	@Override
+	protected HttpHeaders putExtraRequestHeaders(HttpMethod httpMethod, String uri, HttpHeaders headers) throws Exception {
+		headers.set("Content-Type", "text/xml");
+		headers.set("Content-Type", "application/json");
+
+		String sessionID = authenticationCookie.getSessionID();
+		if (sessionID != null && !sessionID.equals(DecoderConstant.AUTHORIZED)) {
+			headers.set(DecoderConstant.COOKIE, authenticationCookie.getSessionID());
+		}
+		return super.putExtraRequestHeaders(httpMethod, uri, headers);
+	}
+
+	/**
+	 * Init instance of Authentication Cookie
+	 *
+	 * @return AuthenticationCookie
+	 */
+	protected AuthenticationCookie initAuthenticationCookie() {
+		return new AuthenticationCookie();
+	}
+
+	/**
+	 * @param path url of the request
+	 * @return String full path of the device
+	 */
+	private String buildDeviceFullPath(String path) {
+		Objects.requireNonNull(path);
+
+		return DecoderConstant.HTTPS
+				+ getHost()
+				+ path;
+	}
+
+	/**
+	 * This method is used to retrieve Authentication Cookie (SessionID) by send POST request to http://{IP_Address}/apis/authentication
+	 *
+	 * When there is no Session ID data or having an Exception, The Session ID of Authentication Cookie is going to set with null value
+	 */
+	private void retrieveSessionFromDecoder() {
+		String login = getLogin();
+		String password = getPassword();
+
+		ObjectNode request = JsonNodeFactory.instance.objectNode();
+		request.put(DecoderConstant.USERNAME, login);
```

Comment:
login and password is not reused in this method, no need to assign it to a variable. please directly call 
request.put(DecoderConstant.USERNAME, getLogin());
request.put(DecoderConstant.PASSWORD, getPassword());

Reply:
I fixed it